CMakeLists.txt

```diff
@@ -9,12 +10,177 @@ endif()
+# loaded plugins.
+add_library(zeek_dynamic_plugin_base INTERFACE)
+target_link_libraries(zeek_dynamic_plugin_base
+  INTERFACE
+    $<BUILD_INTERFACE:zeek_internal>)
+
+# On macOS, we need to tell the linker that the modules are allowed to have
+# undefined symbols.
+if (CMAKE_SYSTEM_NAME MATCHES "Darwin")
+    target_link_options(
+        zeek_dynamic_plugin_base
+        INTERFACE
+            -undefined
+            dynamic_lookup
+            -Wl,-bind_at_load)
+endif ()
+
+function(add_zeek_dynamic_plugin_build_interface_include_directories)
+    foreach ( path ${ARGV} )
+        target_include_directories(
+            zeek_dynamic_plugin_base
+            INTERFACE
+                $<BUILD_INTERFACE:${path}>
+        )
+    endforeach ()
+endfunction()
+
+add_zeek_dynamic_plugin_build_interface_include_directories(
+    ${PROJECT_SOURCE_DIR}/src/include
+    ${PROJECT_SOURCE_DIR}/auxil/binpac/lib
+    ${PROJECT_SOURCE_DIR}/auxil/broker/include
+    ${PROJECT_SOURCE_DIR}/auxil/paraglob/include
+    ${PROJECT_SOURCE_DIR}/auxil/rapidjson/include
+    ${CMAKE_BINARY_DIR}/src/include
+    ${CMAKE_BINARY_DIR}/auxil/binpac/lib
+    ${CMAKE_BINARY_DIR}/auxil/broker/include
+)
+
+# Target for bundling the creation of auto-generated files.
+add_custom_target(zeek_autogen_files)
+
+# Define our main targets and place the output files under src (for historic
+# reasons and backwards compatibility).
+if (ZEEK_STANDALONE)
+    add_executable(zeek_exe)
+    target_link_libraries(zeek_exe PRIVATE $<BUILD_INTERFACE:zeek_internal>)
+    add_dependencies(zeek_exe zeek_autogen_files)
+    set_target_properties(zeek_exe PROPERTIES RUNTIME_OUTPUT_NAME zeek)
+    if (NOT CMAKE_RUNTIME_OUTPUT_DIRECTORY)
+        set_target_properties(zeek_exe PROPERTIES RUNTIME_OUTPUT_DIRECTORY src)
+    endif()
+    if (CONAN_EXPORTED)
+        add_library(zeek_lib STATIC)
+    endif ()
+else ()
+    add_library(zeek_lib STATIC)
+endif()
+
+if (TARGET zeek_lib)
+    target_link_libraries(zeek_lib PRIVATE $<BUILD_INTERFACE:zeek_internal>)
+    add_dependencies(zeek_lib zeek_autogen_files)
+    set_target_properties(zeek_lib PROPERTIES RUNTIME_OUTPUT_NAME libzeek)
+    if (NOT CMAKE_LIBRARY_OUTPUT_DIRECTORY)
+        set_target_properties(zeek_lie PROPERTIES LIBRARY_OUTPUT_DIRECTORY src)
+    endif()
+endif()
+
+# Convenience function for adding object libraries to the main target(s).
+function(zeek_add_obj_lib obj_lib_target)
+    foreach (main_target zeek_exe zeek_lib)
+        if (TARGET ${main_target})
+            target_link_libraries(${main_target} PRIVATE ${obj_lib_target})
+        endif ()
+    endforeach()
+endfunction()
+
+# Convenience function for adding an OBJECT library that feeds directly into the
+# main target(s).
+#
+# Usage:
+# zeek_add_subdir_library(
+#   <name>
+#   SOURCES ...
+#   [INCLUDE_DIRS ...]
+#   [DEPENDENCIES ...]
+#   [INTERNAL_DEPENDENCIES ...]
+#   [BIFS ...]
+# )
+function(zeek_add_subdir_library name)
```

Comment:
Certainly a matter of preference. I would keep them there, because they modify the target defined above and moving the functions someplace else IMHO makes it more difficult to follow the CMake source.